libs/http-ext/src/lib/matchers/match-origin/match-origin.ts

```diff
@@ -1,21 +1,32 @@
 import { RequestCondition } from '../../plugin';
+import { findMatcherOrThrow } from './find-matcher-or-throw';
 import { getOrigin } from './get-origin';
-import { matchArrayOrigin } from './match-array-origin';
-import { matchPredicateOrigin, Predicate } from './match-predicate-origin';
-import { matchRegExpOrigin } from './match-reg-exp-origin';
-import { matchStringOrigin } from './match-string-origin';
+import { originArrayMatcher } from './origin-array-matcher';
+import { OriginMatchExpression } from './origin-match-expression';
+import { originPredicateMatcher } from './origin-predicate-matcher';
+import { originRegExpMatcher } from './origin-reg-exp-matcher';
+import { originStringMatcher } from './origin-string-matcher';
 
-export type Matcher = string | string[] | RegExp | Predicate;
-
-export const matchOrigin = (matcher: Matcher): RequestCondition => ({
-  request
-}): boolean => {
+export const matchOrigin = (
+  matchExpression: OriginMatchExpression
+): RequestCondition => ({ request }): boolean => {
+  /* Extract origin from URL. */
   const origin = getOrigin(request.url);
 
-  return [
-    matchStringOrigin,
-    matchArrayOrigin,
-    matchRegExpOrigin,
-    matchPredicateOrigin
-  ].some(match => match(origin, matcher));
+  /* Find the right matcher. */
+  const matcher = findMatcherOrThrow({
+    matchExpression: matchExpression,
+    matcherList: [
+      originStringMatcher,
+      originArrayMatcher,
+      originRegExpMatcher,
+      originPredicateMatcher
+    ]
+  });
+
+  /* Handle the origin with the right matcher. */
+  return matcher.handle({
+    matchExpression,
+    origin
+  });
```

Comment:
What do you think of this?
The idea is to loop through the matchers and try to find the right one then run only this one or fail because the right matcher couldn't be found.
JavaScript users will be happy to have an error when using something like `matchOrigin({})` for some random reason.

Reply:
Agree with this, it's more explicit.